src/irmin-pack/unix/gc.ml

```diff
@@ -127,11 +117,16 @@ module Make (Args : Args) : S with module Args := Args = struct
   (* Transfer the commit with a different magic. Note that this is modifying
      existing written data. *)
   let transfer_parent_commit ~src ~dst key =
-    let off, len = direct_of_key key in
-    let buffer = Bytes.create len in
-    Fm.Suffix.read_exn src ~off ~len buffer;
-    Bytes.set buffer Hash.hash_size magic_parent;
-    Io.write_exn dst ~off ~len (Bytes.unsafe_to_string buffer)
+    let open Result_syntax in
+    let* off, len =
+      match Irmin_pack.Pack_key.inspect key with
+      | Indexed _ -> Error (`Commit_parent_key_is_indexed (string_of_key key))
```

Comment:
Just to help my understanding: we check the key, which is the key of the parent of a commit; the parent of a commit is itself a commit; the check shows the parent key is indexed (which is normal? all commits are indexed?); but this is now an error? I naively expected the parent would always be indexed. What am I missing?

Reply:
With brand new stores created under irmin 3.x, the only situation where Indexed commit occur is when the user retrieves one from its hash. Irmin 3.x pushes `Commit_v2` on disk, which reference other commits by offset.

Reply:
Yes